frontend/src/pages/Account.tsx

```diff
@@ -1,37 +1,51 @@
-import React, { useMemo, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import useAsyncEffect from 'use-async-effect';
 import { useParams } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
-
-import { useAccount } from 'src/hooks';
+import {
+  fetchAccount,
+  Loading,
+  useAppDispatch,
+  useAppSelector,
+} from 'src/store';
+import {
+  getAccount,
+  getAccountErrorMessage,
+  getAccountLoadingStatus,
+} from 'src/store/selectors/account-selectors';
 import { casperApi } from '../api';
 import { AccountDetailsCard, PageHead, PageWrapper } from '../components';
 
 export const AccountPage: React.FC = () => {
   const { id } = useParams();
   const { t } = useTranslation();
-  const {
-    data: account,
-    error: accountError,
-    isLoading,
-  } = useAccount({
-    accountHashOrPublicKey: id || '',
-  });
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(fetchAccount(id ?? ''));
```

Comment:
I assume we'd never his the empty string, this is just done for TS right?

Reply:
Yes sir!

Reply:
Same as my previous comment about `{}` but I assume its not possible ;)

Reply:
Yup, I believe it's not possible.